src/providers/wfs/oapif/qgsoapifcreatefeaturerequest.cpp

```diff
@@ -0,0 +1,82 @@
+/***************************************************************************
+    qgsoapifcreatefeaturerequest.cpp
+    --------------------------------
+    begin                : March 2023
+    copyright            : (C) 2023 by Even Rouault
+    email                : even.rouault at spatialys.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <nlohmann/json.hpp>
+using namespace nlohmann;
+
+#include "qgslogger.h"
+#include "qgsjsonutils.h"
+#include "qgsoapifcreatefeaturerequest.h"
+#include "qgsoapifprovider.h"
+
+#include <QTransform>
+
+QgsOapifCreateFeatureRequest::QgsOapifCreateFeatureRequest( const QgsDataSourceUri &uri ):
+  QgsBaseNetworkRequest( QgsAuthorizationSettings( uri.username(), uri.password(), uri.authConfigId() ), "OAPIF" )
+{
+}
+
+QString QgsOapifCreateFeatureRequest::createFeature( const QgsOapifSharedData *sharedData, const QgsFeature &f, const QString &contentCrs, bool hasAxisInverted )
+{
+  QgsJsonExporter exporter;
+
+  QgsFeature fModified( f );
+  if ( hasAxisInverted && f.hasGeometry() )
+  {
+    QgsGeometry g = f.geometry();
+    g.transform( QTransform( 0, 1, 1, 0, 0, 0 ) );
```

Comment:
A bit more optimised:

```suggestion
        g.get()->swapXy();
```

Reply:
done